### PR TITLE
OADP-1844: Validation to ensure vsm plugin is removed from DPA if DM is false

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**

### CODE REVIEW

File: controllers/validator.go [hunk 1]
```diff
@@ -52,7 +52,15 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
...
 	}
-	//
+
+	// check if the VSM plugin is specified or not
+	VSMPluginPresent := false
+	for _, plugin := range dpa.Spec.Configuration.Velero.DefaultPlugins {
+		if plugin == oadpv1alpha1.DefaultPluginVSM {
+			VSMPluginPresent = true
+		}
+	}
```
Comment (kaovilai) [inline controllers/validator.go:62]: Why don't we do this closer to line 78 where var is used?

File: controllers/validator.go [hunk 2]
```diff
@@ -62,19 +70,15 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
...
-			}
-		}
-
 		if !VSMPluginPresent {
 			return false, errors.New("datamover feature is enabled, need to specify vsm as a default plugin in DataProtectionApplication CR")
 		}
 	}
 
+	if !r.checkIfDataMoverIsEnabled(&dpa) && VSMPluginPresent {
+		return false, errors.New("datamover feature is disabled, need to remove vsm as a default plugin in DataProtectionApplication CR")
```
Comment (eemcmullan) [inline controllers/validator.go:79]: could possibly shorten this to something like "datamover is disabled-remove vsm as a default plugin"

File: controllers/validator.go (hunk 1 above)
Comment (kaovilai) [reply controllers/validator.go:62]: icic

**Review by weshayutin — APPROVED** [review]
